pyclesperanto_prototype/_tier0/_pycl.py

```diff
@@ -340,13 +340,49 @@ def __ipow__(x1, x2):
             from .._tier1 import power_images
             return power_images(temp, x2, x1)
 
-    def __getitem__(self, index):
-        try:
-            return super().__getitem__(index)
+    def __setitem__(self, index, value):
+        if isinstance(index, tuple) and isinstance(index[0], np.ndarray):
```

Comment:
```suggestion
        if isinstance(index, tuple) and index and isinstance(index[0], np.ndarray):
```

(potential index error otherwise)